torch/fx/_pytree.py

```diff
@@ -1,6 +1,6 @@
 # mypy: allow-untyped-defs
 from collections import namedtuple
-from typing import Any, Callable, Dict, List, NamedTuple, Optional, Tuple, Type
+from typing import Any, Callable, Dict, List, NamedTuple, Optional, Tuple, Type, TypeVar
```

Comment:
We can use `typing_extensions.NamedTuple`:

```python
from typing_extensions import NamedTuple

def _namedtuple_flatten_spec(d: NamedTuple[_T], spec: TreeSpec) -> List[_T]: ...
```

Reply:
Oh the original did not have generics?

Reply:
@XuehaiPan This specific behavior doesn't seem to have been backedported to TypingExtensions yet then: https://typing-extensions.readthedocs.io/en/latest/#typing_extensions.NamedTuple

Reply:
@XuehaiPan It does support Generics, but I am not sure it supports doing a Generics attached to NamedTuple objects quite like this